src/main/java/duke/Duke.java

```diff
@@ -48,58 +48,55 @@ protected String getResponse(String input) {
             responseMessage = taskList.taskMessage;
         } else if (command.equals("done")) {
             int index = Integer.parseInt(input.split(" ")[1]) - 1;
-            if (index < tasks.size() && index >= 0) {
-                taskList.markTaskDone(index);
-                storage.writeToFile(tasks);
-                responseMessage = taskList.taskMessage;
-            } else {
-                responseMessage = ui.taskNotExist();
-            }
+            if (index >= tasks.size() || index < 0)
+                return ui.taskNotExist();
+            taskList.markTaskDone(index);
```

Comment:
To clarify the logic here, you can use else statement although it is understandable still.